applications/IgaApplication/custom_conditions/base_discrete_condition.cpp

```diff
@@ -0,0 +1,344 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Tobias Teschemacher
+//
+
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "custom_conditions/base_discrete_condition.h"
+
+
+namespace Kratos
+{
+    void BaseDiscreteCondition::EquationIdVector(
+        EquationIdVectorType& rResult,
+        ProcessInfo& rCurrentProcessInfo
+    )
+    {
+        KRATOS_TRY;
+
+        KRATOS_ERROR_IF(DofsPerNode() != 3) << "Calling BaseDiscreteCondition::EquationIdVector. Derived Condition does not has 3 DoFs." << std::endl;
+
+        const int number_of_control_points = NumberOfNodes();
+
+        if (rResult.size() != 3 * number_of_control_points)
+            rResult.resize(3 * number_of_control_points, false);
+
+        const unsigned int pos = this->GetGeometry()[0].GetDofPosition(DISPLACEMENT_X);
+
+        for (unsigned int i = 0; i < number_of_control_points; ++i) {
+            const unsigned int index = i * 3;
+            rResult[index]     = GetGeometry()[i].GetDof(DISPLACEMENT_X, pos).EquationId();
+            rResult[index + 1] = GetGeometry()[i].GetDof(DISPLACEMENT_Y, pos + 1).EquationId();
+            rResult[index + 2] = GetGeometry()[i].GetDof(DISPLACEMENT_Z, pos + 2).EquationId();
+        }
+
+        KRATOS_CATCH("")
+    }
+
+    void BaseDiscreteCondition::GetDofList(
+        DofsVectorType& rElementalDofList,
+        ProcessInfo& rCurrentProcessInfo
+    )
+    {
+        KRATOS_TRY;
+
+        KRATOS_ERROR_IF(DofsPerNode() != 3) << "Calling BaseDiscreteCondition::GetDofList. Derived Condition does not has 3 DoFs." << std::endl;
+
+        const int number_of_control_points = NumberOfNodes();
+
+        rElementalDofList.resize(0);
+        rElementalDofList.reserve(3 * number_of_control_points);
+
+        for (unsigned int i = 0; i < number_of_control_points; ++i) {
+            rElementalDofList.push_back(GetGeometry()[i].pGetDof(DISPLACEMENT_X));
+            rElementalDofList.push_back(GetGeometry()[i].pGetDof(DISPLACEMENT_Y));
+            rElementalDofList.push_back(GetGeometry()[i].pGetDof(DISPLACEMENT_Z));
```

Comment:
If performance at this point is so important: would `resize` + `[i]` not be better?